### PR TITLE
RDCC-1905: Move Idempotent logic to library

### PR DESCRIPTION
**Pull request by AdilHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1905

### Change description ###

Move Idempotent logic to library

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/data/ingestion/DataIngestionLibraryRunner.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @Value("${idempotent-flag-ingestion}")
+    boolean isIdempotentFlagEnabled;
+
+    public void run(Job job, JobParameters params) throws Exception {
+
+        if (isIdempotentFlagEnabled) {
+            if (FALSE.equals(auditServiceImpl.isAuditingCompleted())) {
+                log.info("Data Ingestion Library running first time for a day::");
+                jobLauncher.run(job, params);
+                log.info("Data Ingestion Library job run completed::");
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/data/ingestion/DataIngestionLibraryRunner.java]: I think you don't need else ...rather put that line without else and return from above code

Comment (AdilHMCTS) [reply src/main/java/uk/gov/hmcts/reform/data/ingestion/DataIngestionLibraryRunner.java]: Addressed here: https://github.com/hmcts/data-ingestion-lib/pull/12/commits/cd1d80fcfe7d9ba72d761b419e5baf2e7069de1e

File: src/main/java/uk/gov/hmcts/reform/data/ingestion/DataIngestionLibraryRunner.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            } else {
+                log.info("no run of Data Ingestion Library as it has ran for the day::");
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/data/ingestion/DataIngestionLibraryRunner.java]: may be we can optimize


            if (isIdempotentFlagEnabled && TRUE.equals(auditServiceImpl.isAuditingCompleted())) {
              log.info("no run of Data Ingestion Library as it has ran for the day::");
              return;
}
       log.info("Data Ingestion Library job starts::");
       jobLauncher.run(job, params);
       log.info("Data Ingestion Library job run completed::");
    }

Comment (AdilHMCTS) [reply src/main/java/uk/gov/hmcts/reform/data/ingestion/DataIngestionLibraryRunner.java]: Addressed here: https://github.com/hmcts/data-ingestion-lib/pull/12/commits/28e7e681273763f6967f730d81b45b1bbe69c0b1

**Review by abhijitdiwan — APPROVED** [review]

**Review by shashank-rastogi — APPROVED** [review]

**Pull request closed without merging**